extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java

```diff
@@ -102,7 +102,7 @@ AdditionalApplicationArchiveMarkerBuildItem marker() {
 
             ClassInfo current = info;
             boolean done = false;
-            boolean streamHandler = false;
+            boolean streamHandler = info.superName().equals(SKILL_STREAM_HANDLER) ? true : false;
```

Comment:
That won't work if you have an intermediary subclass.

Reply:
Ah OK I see we couldn't arrive here with an intermediary class.

Reply:
Yes, in the while loop it doesn't enter the handleRequest() check, due to methods() only returning methods of the subclass.

```
            while (current != null && !done) {
                for (MethodInfo method : current.methods()) {
                    if (method.name().equals("handleRequest")) {

```

It does skip around the while loop a couple of times, though I though for readability I didn't want to add another condition to the while statement -> && !streamHandler

I debated doing this, and decided against in the end to minimise changes.
`            while (current != null && !done &&!streamHandler) {
`


I also though about doing it a more verbose way for readability and to be consistent on the place in the code where the decision is made:

```
            boolean streamHandler = false;
            while (current != null && !done) {
                for (MethodInfo method : current.methods()) {

                    if (current.superName().equals(SKILL_STREAM_HANDLER)) {
                        streamHandler = true;
                        done = true;
                        break;
                    }
                    
                    if (method.name().equals("handleRequest")) {
                        ...

```